modules/tensor_mechanics/include/materials/ComputeEigenStrainBase.h

```diff
@@ -0,0 +1,32 @@
+#ifndef COMPUTEEIGENSTRAINBASE_H
+#define COMPUTEEIGENSTRAINBASE_H
+
+#include "Material.h"
+#include "RankTwoTensor.h"
+#include "ElasticityTensorR4.h"
+#include "RotationTensor.h"
```

Comment:
Neither RotationTensor.h nor ElasticityTensorR4.h should be included from this header as they are not needed here. I'd rather include them from the implementation (.C) file. 
This also goes for RankTwoTensor.h by the way, as no complete class definition is needed for the MaterialProperty template. It would be better to just forward declare the Tensor type!

```
class RankTwoTensor;
```

The reason is that this can reduce recompilations if certain header files are modified. It will also help to keep overall compile times slightly lower (but that might be a rather small effect)